tests/api2/test_300_nfs.py

```diff
@@ -561,6 +588,83 @@ def test_36_check_nfsdir_subtree_behavior(request):
     assert 'insecure' not in parsed[0]['opts'][0]['parameters'], str(parsed)
 
 
+def test_38_check_nfs_service_v4_parameter(request):
+    """
+    This test verifies that toggling the `v4` option generates expected changes
+    in nfs kernel server config.
+    """
+    depends(request, ["pool_04", "ssh_password"], scope="session")
+
+    results = GET("/nfs")
+    assert results.status_code == 200, results.text
+    assert results.json()['v4'] == True, results.text
+
+    s = parse_server_config()
+    assert "-N 4" not in s["RPCNFSDOPTS"], str(s)
+
+    results = PUT("/nfs/", {"v4": False})
+    assert results.status_code == 200, results.text
+
+    s = parse_server_config()
+    assert "-N 4" in s["RPCNFSDOPTS"], str(s)
+
+    results = PUT("/nfs/", {"v4": True})
+    assert results.status_code == 200, results.text
+
+    s = parse_server_config()
+    assert "-N 4" not in s["RPCNFSDOPTS"], str(s)
+
+
+def test_39_check_nfs_service_udp_parameter(request):
```

Comment:
RFC5661 says:

"_Where an NFSv4.1 implementation supports operation over the IP network protocol, any transport used between NFS and IP MUST be among the IETF-approved congestion control transport protocols.  At the time this document was written, the only two transports that had the above attributes were TCP and the Stream Control Transmission Protocol (SCTP)._ "

Would it make more sense for us to test that if UDP is specified for an NFSv4.x share, we say something about the config not being valid ? Or some such ?

Reply:
That's a good idea. I think we need to convert `v4` to be a RO parameter (for compatibility) and basically allow users to select which protocols they want to support... i.e. allow disabling NFSv3 if they so desire.